From 4b619ec31d77b6d218df11c4ddce8ecb3b3f80d8 Mon Sep 17 00:00:00 2001
From: Leon Anavi <leon.anavi@konsulko.com>
Date: Wed, 9 Jul 2025 08:22:26 +0000
Subject: [PATCH] rpi.env: simplify booting and add dual rootfs

Apply Victron Energy changes from include/configs/rpi.h to the
new text environment format.

Upstream-Status: Inappropriate [Venus specific]

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>
---
 board/raspberrypi/rpi/rpi.env | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/board/raspberrypi/rpi/rpi.env b/board/raspberrypi/rpi/rpi.env
index 30228285edd..f8bd88e79c8 100644
--- a/board/raspberrypi/rpi/rpi.env
+++ b/board/raspberrypi/rpi/rpi.env
@@ -61,13 +61,7 @@ dfu_alt_info+=zImage fat 0 1
  * only 64M, the remaining 25M starting at 0x02700000 should allow quite
  * large initrds before they start colliding with U-Boot.
  */
-#ifdef CONFIG_ARM64
-fdt_high=ffffffffffffffff
-initrd_high=ffffffffffffffff
-#else
-fdt_high=ffffffff
-initrd_high=ffffffff
-#endif
+fdt_high=3000000
 kernel_addr_r=0x00080000
 scriptaddr=0x02400000
 pxefile_addr_r=0x02500000
@@ -75,3 +69,9 @@ fdt_addr_r=0x02600000
 ramdisk_addr_r=0x02700000
 
 boot_targets=mmc usb pxe dhcp
+
+set_args=fdt addr ${fdt_addr} && fdt get value bootargs /chosen bootargs && setenv bootargs "${bootargs} root=${mmcroot} rootwait"
+set_root=if test "${version}" = 2; then setenv bootpart 0:3; setenv mmcroot /dev/mmcblk0p3; else setenv bootpart 0:2; setenv mmcroot /dev/mmcblk0p2; fi
+load=ext2load mmc ${bootpart} ${kernel_addr_r} /boot/Image
+do_boot=booti ${kernel_addr_r} - ${fdt_addr}
+bootcmd=run set_root set_args load do_boot
-- 
2.44.3

